crates/cast/bin/cmd/mktx.rs

```diff
@@ -0,0 +1,108 @@
+use ethers_providers::Middleware;
+use ethers_signers::Signer;
+use eyre::Result;
+use foundry_cli::{
+    opts::{EthereumOpts, TransactionOpts},
+    utils,
+};
+use foundry_common::types::ToAlloy;
+use foundry_config::Config;
+use std::str::FromStr;
+
+/// CLI arguments for `cast mktx`.
+#[derive(Debug, Parser)]
+pub struct MakeTxArgs {
+    /// The destination of the transaction.
+    ///
+    /// If not provided, you must use `cast mktx --create`.
+    #[clap(value_parser = NameOrAddress::from_str)]
+    to: Option<NameOrAddress>,
+
+    /// The signature of the function to call.
+    sig: Option<String>,
+
+    /// The arguments of the function to call.
+    args: Vec<String>,
+
+    /// Reuse the latest nonce for the sender account.
+    #[clap(long, conflicts_with = "nonce")]
+    resend: bool,
+
+    #[clap(subcommand)]
+    command: Option<MakeTxSubcommands>,
+
+    #[clap(flatten)]
+    tx: TransactionOpts,
+
+    #[clap(flatten)]
+    eth: EthereumOpts,
+}
+
+#[derive(Debug, Parser)]
+pub enum MakeTxSubcommands {
+    /// Use to deploy raw contract bytecode.
+    #[clap(name = "--create")]
+    Create {
+        /// The initialization bytecode of the contract to deploy.
+        code: String,
+
+        /// The signature of the constructor.
+        sig: Option<String>,
+
+        /// The constructor arguments.
+        args: Vec<String>,
+    },
+}
+
+impl MakeTxArgs {
+    pub async fn run(self) -> Result<()> {
+        let MakeTxArgs { to, mut sig, mut args, resend, command, mut tx, eth } = self;
+
+        let code = if let Some(MakeTxSubcommands::Create {
+            code,
+            sig: constructor_sig,
+            args: constructor_args,
+        }) = command
+        {
+            sig = constructor_sig;
+            args = constructor_args;
+            Some(code)
+        } else {
+            None
+        };
+
+        let config = Config::from(&eth);
+        let provider = utils::get_provider(&config)?;
+        let chain = utils::get_chain(config.chain, &provider).await?;
+        let api_key = config.get_etherscan_api_key(Some(chain));
+
+        // Retrieve the signer, and bail if it can't be constructed.
+        let signer = eth.wallet.signer(chain.id()).await?;
+        let from = signer.address();
+
+        tx::validate_from_address(eth.wallet.from, from.to_alloy())?;
+        tx::validate_to_address(&code, &to)?;
+
+        if resend {
+            tx.nonce = Some(provider.get_transaction_count(from, None).await?.to_alloy());
+        }
+
+        let provider = provider.with_signer(signer);
+
+        let (mut tx, _) =
+            tx::build_tx(&provider, from, to, code, sig, args, tx, chain, api_key).await?;
+
+        // Fill nonce, gas limit, gas price, and max priority fee per gas if needed
+        provider.fill_transaction(&mut tx, None).await?;
+
+        let signature = provider.sign_transaction(&tx, from).await?;
+        let signed_tx = tx.rlp_signed(&signature);
```

Comment:
Since we're actually signing the transaction, this probably should be behind `cast wallet` similar to the existing `cast wallet sign`. What do you think about also renaming this to `cast wallet sign-tx` to make it more explicit? I don't think cast needs to be backwards compatible with seth anymore. 

I could imagine a separate `cast mktx` command that returns a populated (optionally serialized) transaction, without signing it

Reply:
I went back and forth a few times on this and initially implemented this as `cast wallet sign`, but ultimately decided `cast mktx` is better. The `cast wallet` commands don't use `TxBuilder`, don't use an RPC connection, and therefore have no way of filling a transaction with gas info, nonce, ENS resolution, etc. Having it as `cast mktx` allows it to be as close to `cast send` as possible, with the difference being it prints the encoded transaction instead of broadcasting it. You get everything provided by `TxBuilder` if you need it, but can still specify all the options manually if you have no connection to an RPC endpoint.

Reply:
I suppose it can be renamed to something like `cast sign-tx`, but since the behavior is identical to `seth mktx` (which builds and signs transactions) I went with `cast mktx`.

Reply:
I think a good general principle is to implement the UX that's best for the user, even if it's more work because `cast wallet` commands don't yet have `TxBuilder` or RPC connections. Most foundry users haven't used (or heard of) dapptools and seth is no longer maintained, so we should use the best naming/UX regardless of what seth did.

However, IMO the cast UX is already messy and needs breaking change cleanup, so I won't block this PR on trying to keep cast UX clean or changing command names. Just sharing my thoughts, and will defer to you and other foundry maintainers from here :)

Reply:
Yeah, I agree. I wasn't sure what the best place for this was either. Because of how close its logic is to `cast send`, I put it adjacent to where that is. I can rename/move it wherever (though moving it under `cast wallet` will require a few more changes like importing `TxBuilder` and the utility functions in common with `cast send`).